In [1]:
import pandas as pd

In [2]:
data = pd.read_csv('movies.csv')
data

,MOVIES,YEAR,GENRE,RATING,ONE-LINE,STARS,VOTES,RunTime,Gross
0,Blood Red Sky,(2021),"\nAction, Horror, Thriller",6.1,\nA woman with a mysterious illness is forced ...,\n Director:\nPeter Thorwarth\n| \n Star...,"21,062",121.0,NaN
1,Masters of the Universe: Revelation,(2021– ),"\nAnimation, Action, Adventure",5.0,\nThe war for Eternia begins again in what may...,"\n \n Stars:\nChris Wood, \nSara...","17,870",25.0,NaN
2,The Walking Dead,(2010–2022),"\nDrama, Horror, Thriller",8.2,\nSheriff Deputy Rick Grimes wakes up from a c...,"\n \n Stars:\nAndrew Lincoln, \n...","885,805",44.0,NaN
3,Rick and Morty,(2013– ),"\nAnimation, Adventure, Comedy",9.2,\nAn animated series that follows the exploits...,"\n \n Stars:\nJustin Roiland, \n...","414,849",23.0,NaN
4,Army of Thieves,(2021),"\nAction, Crime, Horror",NaN,"\nA prequel, set before the events of Army of ...",\n Director:\nMatthias Schweighöfer\n| \n ...,NaN,NaN,NaN
...,...,...,...,...,...,...,...,...,...
9994,The Imperfects,(2021– ),"\nAdventure, Drama, Fantasy",NaN,\nAdd a Plot\n,\n \n Stars:\nMorgan Taylor Camp...,NaN,NaN,NaN
9995,Arcane,(2021– ),"\nAnimation, Action, Adventure",NaN,\nAdd a Plot\n,\n,NaN,NaN,NaN
9996,Heart of Invictus,(2022– ),"\nDocumentary, Sport",NaN,\nAdd a Plot\n,\n Director:\nOrlando von Einsiedel\n| \n ...,NaN,NaN,NaN
9997,The Imperfects,(2021– ),"\nAdventure, Drama, Fantasy",NaN,\nAdd a Plot\n,\n Director:\nJovanka Vuckovic\n| \n Sta...,NaN,NaN,NaN


In [3]:
data.columns

Index(['MOVIES', 'YEAR', 'GENRE', 'RATING', 'ONE-LINE', 'STARS', 'VOTES',
       'RunTime', 'Gross'],
      dtype='object')

# Gathered insights 

- [] Movies handle extra spaces and duplicates 
- [] year handle missing values which are 644 and make sure 200x - change to 2xxx and in interger dtype
- [] genre handle special chracter andand missing value of 80
- [] Rating it has 1820 missing values handle them make sure it is from 0 to 10 only and in interger dtype
- [] One line make sure it has no extra spaces and unwanted character and in standardized format 
- [] votes it has many missing values which are around 1820 make sure they are in postive numbers and in interger dtype
- [] Runtime it has 2958 missing values so better handle them and it should be non in negative 
- [] Fross it is better remove tha columsn as it has most of the missing values 

 - []MOVIES
    Strip extra spaces. Drop the 431 exact duplicate rows.
- []    YEAR
Pull just the year number out using regex, since the raw values look like (2021) or (2010–2022). Watch out for entries starting with (I) or (II) — don't let your regex break on those.

- [] GENRE
Strip the leading/trailing whitespace and newline characters. Handle the 80 missing values however you like (drop or fill with "Unknown").

- [] RATING
Keep it as decimal numbers (don't convert to whole integers — you'd lose real precision like 8.7 becoming 9). For the 1,820 missing values: don't drop those rows. Either fill them in with the median rating, or just leave them blank — either is fine for now.

- [] ONE-LINE
Strip the leading \n and extra whitespace.

- [] STARS
Split this into two separate columns — Director and Stars — since right now they're jammed together as one messy text blob with labels and newlines.

- [] VOTES
Remove the commas from numbers like 21,062 and convert to plain integers. Same as Rating — don't drop the 1,820 missing rows, just fill or leave them blank.

- [] RUNTIME
Before deciding anything, check if the really long runtimes (300+ minutes) belong to TV shows vs movies — they likely mean different things. Then fill in the missing values with the median.

- [] GROSS
Just drop this whole column — over 95% of it is missing.

In [4]:
data.isna().sum()

MOVIES         0
YEAR         644
GENRE         80
RATING      1820
ONE-LINE       0
STARS          0
VOTES       1820
RunTime     2958
Gross       9539
dtype: int64

In [5]:
df = data.copy()

In [6]:
df.head()

,MOVIES,YEAR,GENRE,RATING,ONE-LINE,STARS,VOTES,RunTime,Gross
0,Blood Red Sky,(2021),"\nAction, Horror, Thriller",6.1,\nA woman with a mysterious illness is forced ...,\n Director:\nPeter Thorwarth\n| \n Star...,"21,062",121.0,NaN
1,Masters of the Universe: Revelation,(2021– ),"\nAnimation, Action, Adventure",5.0,\nThe war for Eternia begins again in what may...,"\n \n Stars:\nChris Wood, \nSara...","17,870",25.0,NaN
2,The Walking Dead,(2010–2022),"\nDrama, Horror, Thriller",8.2,\nSheriff Deputy Rick Grimes wakes up from a c...,"\n \n Stars:\nAndrew Lincoln, \n...","885,805",44.0,NaN
3,Rick and Morty,(2013– ),"\nAnimation, Adventure, Comedy",9.2,\nAn animated series that follows the exploits...,"\n \n Stars:\nJustin Roiland, \n...","414,849",23.0,NaN
4,Army of Thieves,(2021),"\nAction, Crime, Horror",NaN,"\nA prequel, set before the events of Army of ...",\n Director:\nMatthias Schweighöfer\n| \n ...,NaN,NaN,NaN


In [7]:
df.duplicated().sum()

np.int64(431)

In [8]:
df = df.drop_duplicates()

In [9]:
df.describe()

,RATING,RunTime
count,8168.000000,7008.000000
mean,6.919699,68.874144
std,1.219970,47.278890
min,1.100000,1.000000
25%,6.200000,37.000000
50%,7.100000,60.000000
75%,7.800000,95.000000
max,9.900000,853.000000


In [10]:
df.columns

Index(['MOVIES', 'YEAR', 'GENRE', 'RATING', 'ONE-LINE', 'STARS', 'VOTES',
       'RunTime', 'Gross'],
      dtype='object')

MOVIES
    Strip extra spaces. Drop the 431 exact duplicate rows.

In [11]:
df['MOVIES'] = df['MOVIES'].str.strip()

In [12]:
df[df['MOVIES'].duplicated(keep=False)].sort_values('MOVIES')

,MOVIES,YEAR,GENRE,RATING,ONE-LINE,STARS,VOTES,RunTime,Gross
8375,13 Reasons Why,(2017–2020),"\nDrama, Mystery, Thriller",5.6,\nAs the school gears up for the Love Is Love ...,\n Director:\nMichael Sucsy\n| \n Stars:...,"1,635",59.0,NaN
8395,13 Reasons Why,(2017–2020),"\nDrama, Mystery, Thriller",6.8,\nWhen the dean begins a new investigation and...,\n Director:\nTommy Lohmann\n| \n Stars:...,"1,640",60.0,NaN
7423,13 Reasons Why,(2017–2020),"\nDrama, Mystery, Thriller",5.7,"\nAfter an incident at school, Clay thinks som...",\n Director:\nRussell Mulcahy\n| \n Star...,"1,597",58.0,NaN
8396,13 Reasons Why,(2017–2020),"\nDrama, Mystery, Thriller",7.7,\nStrengthened by the struggles they've endure...,\n Director:\nBrian Yorkey\n| \n Stars:\...,"2,312",98.0,NaN
7378,13 Reasons Why,(2017–2020),"\nDrama, Mystery, Thriller",5.7,"\nThe police question Tyler about the guns, le...",\n Director:\nRussell Mulcahy\n| \n Star...,"1,798",59.0,NaN
...,...,...,...,...,...,...,...,...,...
9289,Ófærð,(2015– ),"\nCrime, Drama, Mystery",NaN,\nEntrapped will pick up two years after the s...,\n Director:\nBaltasar Kormákur\n| \n St...,NaN,NaN,NaN
8457,ÜberWeihnachten,(2020),"\nComedy, Drama, Romance",7.2,\nThe Most Wonderful Christmas of the Year - t...,"\n \n Stars:\nLuke Mockridge, \n...",59,45.0,NaN
8455,ÜberWeihnachten,(2020),"\nComedy, Drama, Romance",7.2,"\nHome Is Where the Tree Is - Bastian, an aspi...","\n \n Stars:\nLuke Mockridge, \n...",65,48.0,NaN
4160,ÜberWeihnachten,(2020),"\nComedy, Drama, Romance",6.7,\nBastian returns home for the holidays to lea...,"\n \n Stars:\nLuke Mockridge, \n...","1,700",48.0,NaN


In [13]:
df['MOVIES'].duplicated().sum()

np.int64(3145)

In [14]:
df['MOVIES'].isna().sum()

np.int64(0)

YEAR
Pull just the year number out using regex, since the raw values look like (2021) or (2010–2022). Watch out for entries starting with (I) or (II) — don't let your regex break on those.


In [15]:
df['YEAR']

0            (2021)
1          (2021– )
2       (2010–2022)
3          (2013– )
4            (2021)
           ...     
9993       (2022– )
9995       (2021– )
9996       (2022– )
9997       (2021– )
9998       (2021– )
Name: YEAR, Length: 9568, dtype: object

In [16]:
df['Year_start'] = df['YEAR'].str.extract(r'(\d{4})')
df['end_year'] = df['YEAR'].str.extract(r'\u2013(\d{4})')
df

,MOVIES,YEAR,GENRE,RATING,ONE-LINE,STARS,VOTES,RunTime,Gross,Year_start,end_year
0,Blood Red Sky,(2021),"\nAction, Horror, Thriller",6.1,\nA woman with a mysterious illness is forced ...,\n Director:\nPeter Thorwarth\n| \n Star...,"21,062",121.0,NaN,2021,NaN
1,Masters of the Universe: Revelation,(2021– ),"\nAnimation, Action, Adventure",5.0,\nThe war for Eternia begins again in what may...,"\n \n Stars:\nChris Wood, \nSara...","17,870",25.0,NaN,2021,NaN
2,The Walking Dead,(2010–2022),"\nDrama, Horror, Thriller",8.2,\nSheriff Deputy Rick Grimes wakes up from a c...,"\n \n Stars:\nAndrew Lincoln, \n...","885,805",44.0,NaN,2010,2022
3,Rick and Morty,(2013– ),"\nAnimation, Adventure, Comedy",9.2,\nAn animated series that follows the exploits...,"\n \n Stars:\nJustin Roiland, \n...","414,849",23.0,NaN,2013,NaN
4,Army of Thieves,(2021),"\nAction, Crime, Horror",NaN,"\nA prequel, set before the events of Army of ...",\n Director:\nMatthias Schweighöfer\n| \n ...,NaN,NaN,NaN,2021,NaN
...,...,...,...,...,...,...,...,...,...,...,...
9993,Totenfrau,(2022– ),"\nDrama, Thriller",NaN,\nAdd a Plot\n,\n Director:\nNicolai Rohde\n| \n Stars:...,NaN,NaN,NaN,2022,NaN
9995,Arcane,(2021– ),"\nAnimation, Action, Adventure",NaN,\nAdd a Plot\n,\n,NaN,NaN,NaN,2021,NaN
9996,Heart of Invictus,(2022– ),"\nDocumentary, Sport",NaN,\nAdd a Plot\n,\n Director:\nOrlando von Einsiedel\n| \n ...,NaN,NaN,NaN,2022,NaN
9997,The Imperfects,(2021– ),"\nAdventure, Drama, Fantasy",NaN,\nAdd a Plot\n,\n Director:\nJovanka Vuckovic\n| \n Sta...,NaN,NaN,NaN,2021,NaN


In [17]:
df.loc[df['YEAR'].str.contains(r'(\d{4}\u2013\s*\))',na=False)].sort_values('YEAR')
df['is_ongoing'] = df['YEAR'].str.contains(r'\d{4}\u2013\s*\)',na=False)

/var/folders/qw/_nk9z1hd74q9vjp80k85r5l80000gn/T/ipykernel_60558/641944660.py:1: UserWarning: This pattern is interpreted as a regular expression, and has match groups. To actually get the groups, use str.extract.
  df.loc[df['YEAR'].str.contains(r'(\d{4}\u2013\s*\))',na=False)].sort_values('YEAR')


In [18]:
df

,MOVIES,YEAR,GENRE,RATING,ONE-LINE,STARS,VOTES,RunTime,Gross,Year_start,end_year,is_ongoing
0,Blood Red Sky,(2021),"\nAction, Horror, Thriller",6.1,\nA woman with a mysterious illness is forced ...,\n Director:\nPeter Thorwarth\n| \n Star...,"21,062",121.0,NaN,2021,NaN,False
1,Masters of the Universe: Revelation,(2021– ),"\nAnimation, Action, Adventure",5.0,\nThe war for Eternia begins again in what may...,"\n \n Stars:\nChris Wood, \nSara...","17,870",25.0,NaN,2021,NaN,True
2,The Walking Dead,(2010–2022),"\nDrama, Horror, Thriller",8.2,\nSheriff Deputy Rick Grimes wakes up from a c...,"\n \n Stars:\nAndrew Lincoln, \n...","885,805",44.0,NaN,2010,2022,False
3,Rick and Morty,(2013– ),"\nAnimation, Adventure, Comedy",9.2,\nAn animated series that follows the exploits...,"\n \n Stars:\nJustin Roiland, \n...","414,849",23.0,NaN,2013,NaN,True
4,Army of Thieves,(2021),"\nAction, Crime, Horror",NaN,"\nA prequel, set before the events of Army of ...",\n Director:\nMatthias Schweighöfer\n| \n ...,NaN,NaN,NaN,2021,NaN,False
...,...,...,...,...,...,...,...,...,...,...,...,...
9993,Totenfrau,(2022– ),"\nDrama, Thriller",NaN,\nAdd a Plot\n,\n Director:\nNicolai Rohde\n| \n Stars:...,NaN,NaN,NaN,2022,NaN,True
9995,Arcane,(2021– ),"\nAnimation, Action, Adventure",NaN,\nAdd a Plot\n,\n,NaN,NaN,NaN,2021,NaN,True
9996,Heart of Invictus,(2022– ),"\nDocumentary, Sport",NaN,\nAdd a Plot\n,\n Director:\nOrlando von Einsiedel\n| \n ...,NaN,NaN,NaN,2022,NaN,True
9997,The Imperfects,(2021– ),"\nAdventure, Drama, Fantasy",NaN,\nAdd a Plot\n,\n Director:\nJovanka Vuckovic\n| \n Sta...,NaN,NaN,NaN,2021,NaN,True


(I) or (II)

In [19]:
df[df['YEAR'].str.contains('I |II',na=False)]

,MOVIES,YEAR,GENRE,RATING,ONE-LINE,STARS,VOTES,RunTime,Gross,Year_start,end_year,is_ongoing
84,Heartland,(II) (2007– ),"\nDrama, Family",8.4,"\nA multi-generational saga set in Alberta, Ca...","\n \n Stars:\nAmber Marshall, \n...","11,764",45.0,NaN,2007,NaN,True
200,Awake,(III) (2021),"\nAction, Adventure, Drama",4.8,\nAfter a devastating global event wiped out a...,\n Director:\nMark Raso\n| \n Stars:\nGi...,"16,262",96.0,NaN,2021,NaN,False
204,Arrival,(II) (2016),"\nDrama, Sci-Fi",7.9,\nA linguist works with the military to commun...,\n Director:\nDenis Villeneuve\n| \n Sta...,"623,912",116.0,$100.55M,2016,NaN,False
340,Hell or High Water,(II) (2016),"\nAction, Crime, Drama",7.6,\nA divorced father and his ex-con older broth...,\n Director:\nDavid Mackenzie\n| \n Star...,"211,816",102.0,$26.86M,2016,NaN,False
434,Manhunt,(II) (2017– ),"\nBiography, Crime, Drama",8.1,\nAn in-depth look at how an FBI profiler help...,"\n \n Stars:\nGethin Anthony, \n...","54,164",60.0,NaN,2017,NaN,True
...,...,...,...,...,...,...,...,...,...,...,...,...
6857,The Pentaverate,(II),\nComedy,NaN,\nAdd a Plot\n,\n Director:\nTim Kirkby\n| \n Stars:\nN...,NaN,NaN,NaN,NaN,NaN,False
9461,True Story,(III),"\nComedy, Drama",NaN,\nAdd a Plot\n,\n Director:\nStephen Williams\n| \n Sta...,NaN,NaN,NaN,NaN,NaN,False
9649,True Story,(III),"\nComedy, Drama",NaN,\nAdd a Plot\n,\n Director:\nStephen Williams\n| \n Sta...,NaN,NaN,NaN,NaN,NaN,False
9652,True Story,(III),"\nComedy, Drama",NaN,\nAdd a Plot\n,\n Director:\nHanelle M. Culpepper\n| \n ...,NaN,NaN,NaN,NaN,NaN,False


GENRE
Strip the leading/trailing whitespace and newline characters. Handle the 80 missing values however you like (drop or fill with "Unknown").


In [20]:
df= df.dropna(subset='GENRE')
df[df['GENRE'].isna()]

,MOVIES,YEAR,GENRE,RATING,ONE-LINE,STARS,VOTES,RunTime,Gross,Year_start,end_year,is_ongoing


In [21]:
df.shape

(9490, 12)

In [22]:
df['GENRE'] = df['GENRE'].str.strip()

/var/folders/qw/_nk9z1hd74q9vjp80k85r5l80000gn/T/ipykernel_60558/974377861.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df['GENRE'] = df['GENRE'].str.strip()


In [23]:
df['GENRE'].str.contains(r'^\s|\s$',regex=True,na=False).sum()

np.int64(0)

RATING
Keep it as decimal numbers (don't convert to whole integers — you'd lose real precision like 8.7 becoming 9). For the 1,820 missing values: don't drop those rows. Either fill them in with the median rating, or just leave them blank — either is fine for now.

In [24]:
df['RATING'].dtype

dtype('float64')

In [25]:
df['RATING'].isna().sum()
df[df['RATING'] >= 10]

,MOVIES,YEAR,GENRE,RATING,ONE-LINE,STARS,VOTES,RunTime,Gross,Year_start,end_year,is_ongoing


In [26]:
df['RATING'].min(), df['RATING'].max()

(1.1, 9.9)

In [27]:
df['RATING'] = df['RATING'].fillna(df['RATING'].median())

/var/folders/qw/_nk9z1hd74q9vjp80k85r5l80000gn/T/ipykernel_60558/3222467561.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df['RATING'] = df['RATING'].fillna(df['RATING'].median())


In [28]:
df['RATING'].isnull().sum()

np.int64(0)

 ONE-LINE
Strip the leading \n and extra whitespace.

In [29]:
df['ONE-LINE']

0       \nA woman with a mysterious illness is forced ...
1       \nThe war for Eternia begins again in what may...
2       \nSheriff Deputy Rick Grimes wakes up from a c...
3       \nAn animated series that follows the exploits...
4       \nA prequel, set before the events of Army of ...
                              ...                        
9993                                       \nAdd a Plot\n
9995                                       \nAdd a Plot\n
9996                                       \nAdd a Plot\n
9997                                       \nAdd a Plot\n
9998                                       \nAdd a Plot\n
Name: ONE-LINE, Length: 9490, dtype: object

In [30]:
df['ONE-LINE'] = df['ONE-LINE'].str.strip()

/var/folders/qw/_nk9z1hd74q9vjp80k85r5l80000gn/T/ipykernel_60558/1667020961.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df['ONE-LINE'] = df['ONE-LINE'].str.strip()


In [31]:
df['ONE-LINE'].isna().sum()

df[df['ONE-LINE'].str.contains(r'[^A-Za-z0-9\s]' , regex= True)].sort_values('ONE-LINE')

,MOVIES,YEAR,GENRE,RATING,ONE-LINE,STARS,VOTES,RunTime,Gross,Year_start,end_year,is_ongoing
9486,The Drew Barrymore Show,(2020– ),Talk-Show,7.1,"""All American"" star Taye Diggs will joins Drew...",\n Director:\nAdam Heydt\n| \n Stars:\nD...,NaN,44.0,NaN,2020,NaN,True
3249,Todas las pecas del mundo,(2019),"Comedy, Romance, Sport",5.2,"""All the Freckles in the World,"" set in 1990s ...",\n Director:\nYibran Asuad\n| \n Stars:\...,"1,142",91.0,NaN,2019,NaN,False
4746,OCTB,(2017– ),"Crime, Drama",7.2,"""An undercover detective crosses paths with fa...",\n \n Stars:\nBrian Thomas Burre...,161,24.0,NaN,2017,NaN,True
4515,Areumdaun Sesang,(2019– ),"Drama, Mystery",8.4,"""Beautiful World,"" tells of an incident of sch...","\n \n Stars:\nHee-soon Park, \nJ...",94,60.0,NaN,2019,NaN,True
9344,The Drew Barrymore Show,(2020– ),Talk-Show,7.1,"""Cobra Kai"" star Ralph Macchio will stops by t...",\n Director:\nAdam Heydt\n| \n Stars:\nD...,NaN,44.0,NaN,2020,NaN,True
...,...,...,...,...,...,...,...,...,...,...,...,...
5494,We Lost Our Human,(2022 TV Special),"Animation, Adventure, Comedy",7.1,the perspective of a self-involved cat (Pud) o...,\n Director:\nCurtis Lelash\n| \n Stars:...,NaN,NaN,NaN,2022,NaN,False
7181,Go! Vive a Tu Manera,(2019),"Comedy, Musical, Romance",8.0,Álvaro puts together a surprise dinner for Mía...,"\n \n Stars:\nDarcy Rose Byrnes,...",7,NaN,NaN,2019,NaN,False
7185,Go! Vive a Tu Manera,(2019),"Comedy, Musical, Romance",8.1,Álvaro struggles to convince Juanma to play ba...,"\n \n Stars:\nDarcy Rose Byrnes,...",7,NaN,NaN,2019,NaN,False
4075,Çok Filim Hareketler Bunlar,(2010),Comedy,4.4,Çok film hareketler bunlar They are a 2010 com...,\n Director:\nOzan Açiktan\n| \n Stars:\...,"4,598",134.0,NaN,2010,NaN,False


STARS
Split this into two separate columns — Director and Stars — since right now they're jammed together as one messy text blob with labels and newlines.

In [32]:
df['STARS'].isna().sum()

np.int64(0)

In [33]:
df[df['STARS'].str.contains(r'Director:\s*(.*?)\s*\|',na=False)]


/var/folders/qw/_nk9z1hd74q9vjp80k85r5l80000gn/T/ipykernel_60558/3434378486.py:1: UserWarning: This pattern is interpreted as a regular expression, and has match groups. To actually get the groups, use str.extract.
  df[df['STARS'].str.contains(r'Director:\s*(.*?)\s*\|',na=False)]


,MOVIES,YEAR,GENRE,RATING,ONE-LINE,STARS,VOTES,RunTime,Gross,Year_start,end_year,is_ongoing
0,Blood Red Sky,(2021),"Action, Horror, Thriller",6.1,A woman with a mysterious illness is forced in...,\n Director:\nPeter Thorwarth\n| \n Star...,"21,062",121.0,NaN,2021,NaN,False
4,Army of Thieves,(2021),"Action, Crime, Horror",7.1,"A prequel, set before the events of Army of th...",\n Director:\nMatthias Schweighöfer\n| \n ...,NaN,NaN,NaN,2021,NaN,False
6,The Last Letter from Your Lover,(2021),"Drama, Romance",6.8,A pair of interwoven stories set in the past a...,\n Director:\nAugustine Frizzell\n| \n S...,"5,283",110.0,NaN,2021,NaN,False
10,Gunpowder Milkshake,(2021),"Action, Adventure, Thriller",6.0,Three generations of women fight back against ...,\n Director:\nNavot Papushado\n| \n Star...,"17,989",114.0,NaN,2021,NaN,False
12,Fear Street: 1994,(2021),"Drama, Horror, Mystery",6.2,A circle of teenage friends accidentally encou...,\n Director:\nLeigh Janiak\n| \n Stars:\...,"50,148",107.0,NaN,2021,NaN,False
...,...,...,...,...,...,...,...,...,...,...,...,...
9983,The Last Kingdom,(2015– ),"Action, Drama, History",7.1,Add a Plot,\n Director:\nPaul Wilmshurst\n| \n Star...,NaN,NaN,NaN,2015,NaN,True
9993,Totenfrau,(2022– ),"Drama, Thriller",7.1,Add a Plot,\n Director:\nNicolai Rohde\n| \n Stars:...,NaN,NaN,NaN,2022,NaN,True
9996,Heart of Invictus,(2022– ),"Documentary, Sport",7.1,Add a Plot,\n Director:\nOrlando von Einsiedel\n| \n ...,NaN,NaN,NaN,2022,NaN,True
9997,The Imperfects,(2021– ),"Adventure, Drama, Fantasy",7.1,Add a Plot,\n Director:\nJovanka Vuckovic\n| \n Sta...,NaN,NaN,NaN,2021,NaN,True


In [34]:
df

,MOVIES,YEAR,GENRE,RATING,ONE-LINE,STARS,VOTES,RunTime,Gross,Year_start,end_year,is_ongoing
0,Blood Red Sky,(2021),"Action, Horror, Thriller",6.1,A woman with a mysterious illness is forced in...,\n Director:\nPeter Thorwarth\n| \n Star...,"21,062",121.0,NaN,2021,NaN,False
1,Masters of the Universe: Revelation,(2021– ),"Animation, Action, Adventure",5.0,The war for Eternia begins again in what may b...,"\n \n Stars:\nChris Wood, \nSara...","17,870",25.0,NaN,2021,NaN,True
2,The Walking Dead,(2010–2022),"Drama, Horror, Thriller",8.2,Sheriff Deputy Rick Grimes wakes up from a com...,"\n \n Stars:\nAndrew Lincoln, \n...","885,805",44.0,NaN,2010,2022,False
3,Rick and Morty,(2013– ),"Animation, Adventure, Comedy",9.2,An animated series that follows the exploits o...,"\n \n Stars:\nJustin Roiland, \n...","414,849",23.0,NaN,2013,NaN,True
4,Army of Thieves,(2021),"Action, Crime, Horror",7.1,"A prequel, set before the events of Army of th...",\n Director:\nMatthias Schweighöfer\n| \n ...,NaN,NaN,NaN,2021,NaN,False
...,...,...,...,...,...,...,...,...,...,...,...,...
9993,Totenfrau,(2022– ),"Drama, Thriller",7.1,Add a Plot,\n Director:\nNicolai Rohde\n| \n Stars:...,NaN,NaN,NaN,2022,NaN,True
9995,Arcane,(2021– ),"Animation, Action, Adventure",7.1,Add a Plot,\n,NaN,NaN,NaN,2021,NaN,True
9996,Heart of Invictus,(2022– ),"Documentary, Sport",7.1,Add a Plot,\n Director:\nOrlando von Einsiedel\n| \n ...,NaN,NaN,NaN,2022,NaN,True
9997,The Imperfects,(2021– ),"Adventure, Drama, Fantasy",7.1,Add a Plot,\n Director:\nJovanka Vuckovic\n| \n Sta...,NaN,NaN,NaN,2021,NaN,True


In [35]:
df[df['STARS'].str.contains(r'Stars:',na=False)]

,MOVIES,YEAR,GENRE,RATING,ONE-LINE,STARS,VOTES,RunTime,Gross,Year_start,end_year,is_ongoing
0,Blood Red Sky,(2021),"Action, Horror, Thriller",6.1,A woman with a mysterious illness is forced in...,\n Director:\nPeter Thorwarth\n| \n Star...,"21,062",121.0,NaN,2021,NaN,False
1,Masters of the Universe: Revelation,(2021– ),"Animation, Action, Adventure",5.0,The war for Eternia begins again in what may b...,"\n \n Stars:\nChris Wood, \nSara...","17,870",25.0,NaN,2021,NaN,True
2,The Walking Dead,(2010–2022),"Drama, Horror, Thriller",8.2,Sheriff Deputy Rick Grimes wakes up from a com...,"\n \n Stars:\nAndrew Lincoln, \n...","885,805",44.0,NaN,2010,2022,False
3,Rick and Morty,(2013– ),"Animation, Adventure, Comedy",9.2,An animated series that follows the exploits o...,"\n \n Stars:\nJustin Roiland, \n...","414,849",23.0,NaN,2013,NaN,True
4,Army of Thieves,(2021),"Action, Crime, Horror",7.1,"A prequel, set before the events of Army of th...",\n Director:\nMatthias Schweighöfer\n| \n ...,NaN,NaN,NaN,2021,NaN,False
...,...,...,...,...,...,...,...,...,...,...,...,...
9979,Never Have I Ever,(2020– ),Comedy,8.8,As Paxton's mixed messages have Devi questioni...,\n Director:\nLang Fisher\n| \n Stars:\n...,433,NaN,NaN,2020,NaN,True
9983,The Last Kingdom,(2015– ),"Action, Drama, History",7.1,Add a Plot,\n Director:\nPaul Wilmshurst\n| \n Star...,NaN,NaN,NaN,2015,NaN,True
9993,Totenfrau,(2022– ),"Drama, Thriller",7.1,Add a Plot,\n Director:\nNicolai Rohde\n| \n Stars:...,NaN,NaN,NaN,2022,NaN,True
9997,The Imperfects,(2021– ),"Adventure, Drama, Fantasy",7.1,Add a Plot,\n Director:\nJovanka Vuckovic\n| \n Sta...,NaN,NaN,NaN,2021,NaN,True


In [36]:
df['Stars'] = df['STARS'].str.extract(r'Stars:\s*([\s\S]*?)(?:\s*\||$)',expand= False)

/var/folders/qw/_nk9z1hd74q9vjp80k85r5l80000gn/T/ipykernel_60558/1247216532.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df['Stars'] = df['STARS'].str.extract(r'Stars:\s*([\s\S]*?)(?:\s*\||$)',expand= False)


In [37]:
df['Stars'].isnull().sum()

np.int64(1146)

In [38]:
df['Stars'] = df['Stars'].str.strip()

/var/folders/qw/_nk9z1hd74q9vjp80k85r5l80000gn/T/ipykernel_60558/1655028087.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df['Stars'] = df['Stars'].str.strip()


In [39]:
df

,MOVIES,YEAR,GENRE,RATING,ONE-LINE,STARS,VOTES,RunTime,Gross,Year_start,end_year,is_ongoing,Stars
0,Blood Red Sky,(2021),"Action, Horror, Thriller",6.1,A woman with a mysterious illness is forced in...,\n Director:\nPeter Thorwarth\n| \n Star...,"21,062",121.0,NaN,2021,NaN,False,"Peri Baumeister, \nCarl Anton Koch, \nAlexande..."
1,Masters of the Universe: Revelation,(2021– ),"Animation, Action, Adventure",5.0,The war for Eternia begins again in what may b...,"\n \n Stars:\nChris Wood, \nSara...","17,870",25.0,NaN,2021,NaN,True,"Chris Wood, \nSarah Michelle Gellar, \nLena He..."
2,The Walking Dead,(2010–2022),"Drama, Horror, Thriller",8.2,Sheriff Deputy Rick Grimes wakes up from a com...,"\n \n Stars:\nAndrew Lincoln, \n...","885,805",44.0,NaN,2010,2022,False,"Andrew Lincoln, \nNorman Reedus, \nMelissa McB..."
3,Rick and Morty,(2013– ),"Animation, Adventure, Comedy",9.2,An animated series that follows the exploits o...,"\n \n Stars:\nJustin Roiland, \n...","414,849",23.0,NaN,2013,NaN,True,"Justin Roiland, \nChris Parnell, \nSpencer Gra..."
4,Army of Thieves,(2021),"Action, Crime, Horror",7.1,"A prequel, set before the events of Army of th...",\n Director:\nMatthias Schweighöfer\n| \n ...,NaN,NaN,NaN,2021,NaN,False,"Matthias Schweighöfer, \nNathalie Emmanuel, \n..."
...,...,...,...,...,...,...,...,...,...,...,...,...,...
9993,Totenfrau,(2022– ),"Drama, Thriller",7.1,Add a Plot,\n Director:\nNicolai Rohde\n| \n Stars:...,NaN,NaN,NaN,2022,NaN,True,"Felix Klare, \nRomina Küper, \nAnna Maria Mühe..."
9995,Arcane,(2021– ),"Animation, Action, Adventure",7.1,Add a Plot,\n,NaN,NaN,NaN,2021,NaN,True,NaN
9996,Heart of Invictus,(2022– ),"Documentary, Sport",7.1,Add a Plot,\n Director:\nOrlando von Einsiedel\n| \n ...,NaN,NaN,NaN,2022,NaN,True,NaN
9997,The Imperfects,(2021– ),"Adventure, Drama, Fantasy",7.1,Add a Plot,\n Director:\nJovanka Vuckovic\n| \n Sta...,NaN,NaN,NaN,2021,NaN,True,"Morgan Taylor Campbell, \nIñaki Godoy, \nRhian..."


In [40]:
df['Director'] = df['STARS'].str.extract(r'Director:\s*(.*?)\s*\|')
df


/var/folders/qw/_nk9z1hd74q9vjp80k85r5l80000gn/T/ipykernel_60558/1023180953.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df['Director'] = df['STARS'].str.extract(r'Director:\s*(.*?)\s*\|')


,MOVIES,YEAR,GENRE,RATING,ONE-LINE,STARS,VOTES,RunTime,Gross,Year_start,end_year,is_ongoing,Stars,Director
0,Blood Red Sky,(2021),"Action, Horror, Thriller",6.1,A woman with a mysterious illness is forced in...,\n Director:\nPeter Thorwarth\n| \n Star...,"21,062",121.0,NaN,2021,NaN,False,"Peri Baumeister, \nCarl Anton Koch, \nAlexande...",Peter Thorwarth
1,Masters of the Universe: Revelation,(2021– ),"Animation, Action, Adventure",5.0,The war for Eternia begins again in what may b...,"\n \n Stars:\nChris Wood, \nSara...","17,870",25.0,NaN,2021,NaN,True,"Chris Wood, \nSarah Michelle Gellar, \nLena He...",NaN
2,The Walking Dead,(2010–2022),"Drama, Horror, Thriller",8.2,Sheriff Deputy Rick Grimes wakes up from a com...,"\n \n Stars:\nAndrew Lincoln, \n...","885,805",44.0,NaN,2010,2022,False,"Andrew Lincoln, \nNorman Reedus, \nMelissa McB...",NaN
3,Rick and Morty,(2013– ),"Animation, Adventure, Comedy",9.2,An animated series that follows the exploits o...,"\n \n Stars:\nJustin Roiland, \n...","414,849",23.0,NaN,2013,NaN,True,"Justin Roiland, \nChris Parnell, \nSpencer Gra...",NaN
4,Army of Thieves,(2021),"Action, Crime, Horror",7.1,"A prequel, set before the events of Army of th...",\n Director:\nMatthias Schweighöfer\n| \n ...,NaN,NaN,NaN,2021,NaN,False,"Matthias Schweighöfer, \nNathalie Emmanuel, \n...",Matthias Schweighöfer
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
9993,Totenfrau,(2022– ),"Drama, Thriller",7.1,Add a Plot,\n Director:\nNicolai Rohde\n| \n Stars:...,NaN,NaN,NaN,2022,NaN,True,"Felix Klare, \nRomina Küper, \nAnna Maria Mühe...",Nicolai Rohde
9995,Arcane,(2021– ),"Animation, Action, Adventure",7.1,Add a Plot,\n,NaN,NaN,NaN,2021,NaN,True,NaN,NaN
9996,Heart of Invictus,(2022– ),"Documentary, Sport",7.1,Add a Plot,\n Director:\nOrlando von Einsiedel\n| \n ...,NaN,NaN,NaN,2022,NaN,True,NaN,Orlando von Einsiedel
9997,The Imperfects,(2021– ),"Adventure, Drama, Fantasy",7.1,Add a Plot,\n Director:\nJovanka Vuckovic\n| \n Sta...,NaN,NaN,NaN,2021,NaN,True,"Morgan Taylor Campbell, \nIñaki Godoy, \nRhian...",Jovanka Vuckovic


In [41]:
df['Director'] = df['Director'].str.strip()

/var/folders/qw/_nk9z1hd74q9vjp80k85r5l80000gn/T/ipykernel_60558/583335392.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df['Director'] = df['Director'].str.strip()


VOTES
Remove the commas from numbers like 21,062 and convert to plain integers. Same as Rating — don't drop the 1,820 missing rows, just fill or leave them blank.

In [42]:
df['VOTES'].isna().sum()

np.int64(1333)

In [43]:
df['VOTES'] = df['VOTES'].str.replace(',' ,'')

/var/folders/qw/_nk9z1hd74q9vjp80k85r5l80000gn/T/ipykernel_60558/370403169.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df['VOTES'] = df['VOTES'].str.replace(',' ,'')


In [44]:
df['VOTES'] = pd.to_numeric(df['VOTES'])

/var/folders/qw/_nk9z1hd74q9vjp80k85r5l80000gn/T/ipykernel_60558/2111369324.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df['VOTES'] = pd.to_numeric(df['VOTES'])


In [45]:
df['VOTES'].dtype

dtype('float64')

In [46]:
df['VOTES']

0        21062.0
1        17870.0
2       885805.0
3       414849.0
4            NaN
          ...   
9993         NaN
9995         NaN
9996         NaN
9997         NaN
9998         NaN
Name: VOTES, Length: 9490, dtype: float64

In [47]:
df['VOTES'] = df['VOTES'].fillna(df['VOTES'].median())

/var/folders/qw/_nk9z1hd74q9vjp80k85r5l80000gn/T/ipykernel_60558/56371188.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df['VOTES'] = df['VOTES'].fillna(df['VOTES'].median())


In [48]:
df['VOTES'].isna().sum()

np.int64(0)

RUNTIME
Before deciding anything, check if the really long runtimes (300+ minutes) belong to TV shows vs movies — they likely mean different things. Then fill in the missing values with the median.

In [49]:
df['RunTime'].isna().sum()

np.int64(2491)

In [50]:
df[df['RunTime'] >=300]

,MOVIES,YEAR,GENRE,RATING,ONE-LINE,STARS,VOTES,RunTime,Gross,Year_start,end_year,is_ongoing,Stars,Director
60,The Queen's Gambit,(2020),Drama,8.6,"Orphaned at the tender age of nine, prodigious...","\n \n Stars:\nAnya Taylor-Joy, \...",336801.0,395.0,NaN,2020,NaN,False,"Anya Taylor-Joy, \nChloe Pirrie, \nBill Camp, ...",NaN
201,The Haunting of Hill House,(2018),"Drama, Horror, Mystery",8.6,"Flashing between past and present, a fractured...","\n \n Stars:\nMichiel Huisman, \...",195117.0,572.0,NaN,2018,NaN,False,"Michiel Huisman, \nCarla Gugino, \nHenry Thoma...",NaN
207,El inocente,(2021),"Crime, Drama, Mystery",7.9,An accidental killing leads a man down a dark ...,"\n \n Stars:\nMario Casas, \nXav...",19980.0,467.0,NaN,2021,NaN,False,"Mario Casas, \nXavi Sáez, \nSanti Pons, \nSam ...",NaN
331,Unbelievable,(2019),"Crime, Drama",8.4,A teenager is charged with lying about having ...,"\n \n Stars:\nKaitlyn Dever, \nT...",81658.0,385.0,NaN,2019,NaN,False,"Kaitlyn Dever, \nToni Collette, \nMerritt Weve...",NaN
347,Godless,(2017),"Drama, Western",8.3,"In the 1880s American West, murderous outlaw g...","\n \n Stars:\nJack O'Connell, \n...",56997.0,452.0,NaN,2017,NaN,False,"Jack O'Connell, \nMichelle Dockery, \nScoot Mc...",NaN
352,The Haunting of Bly Manor,(2020),"Drama, Horror, Mystery",7.4,"After an au pair’s tragic death, Henry hires a...","\n \n Stars:\nVictoria Pedretti,...",77872.0,494.0,NaN,2020,NaN,False,"Victoria Pedretti, \nOliver Jackson-Cohen, \nA...",NaN
427,Maniac,(2018),"Comedy, Drama, Sci-Fi",7.8,Two strangers are drawn to a mysterious pharma...,"\n \n Stars:\nJonah Hill, \nEmma...",69384.0,386.0,NaN,2018,NaN,False,"Jonah Hill, \nEmma Stone, \nSonoya Mizuno, \nJ...",NaN
439,The Stranger,(I) (2020),"Crime, Drama, Mystery",7.3,Follows the lives of suburban families whose s...,"\n \n Stars:\nRichard Armitage, ...",38697.0,363.0,NaN,2020,NaN,False,"Richard Armitage, \nShaun Dooley, \nSiobhan Fi...",NaN
509,Safe,(2018),"Crime, Drama, Mystery",7.3,"After his teenage daughter goes missing, a wid...","\n \n Stars:\nMichael C. Hall, \...",34984.0,356.0,NaN,2018,NaN,False,"Michael C. Hall, \nAmy James-Kelly, \nAmanda A...",NaN
514,The Last Dance,(2020),"Documentary, Biography, History",9.1,"Charting the rise of the 1990's Chicago Bulls,...","\n \n Stars:\nPhil Jackson, \nMi...",93041.0,491.0,NaN,2020,NaN,False,"Phil Jackson, \nMichael Jordan, \nDavid Aldrid...",NaN


In [51]:
df.columns

Index(['MOVIES', 'YEAR', 'GENRE', 'RATING', 'ONE-LINE', 'STARS', 'VOTES',
       'RunTime', 'Gross', 'Year_start', 'end_year', 'is_ongoing', 'Stars',
       'Director'],
      dtype='object')

In [52]:
df['RunTime'].describe()

count    6999.000000
mean       68.855694
std        47.288931
min         1.000000
25%        37.000000
50%        60.000000
75%        95.000000
max       853.000000
Name: RunTime, dtype: float64

In [53]:
df[df['RunTime'] >= 300][['MOVIES', 'RunTime', 'GENRE']]

,MOVIES,RunTime,GENRE
60,The Queen's Gambit,395.0,Drama
201,The Haunting of Hill House,572.0,"Drama, Horror, Mystery"
207,El inocente,467.0,"Crime, Drama, Mystery"
331,Unbelievable,385.0,"Crime, Drama"
347,Godless,452.0,"Drama, Western"
352,The Haunting of Bly Manor,494.0,"Drama, Horror, Mystery"
427,Maniac,386.0,"Comedy, Drama, Sci-Fi"
439,The Stranger,363.0,"Crime, Drama, Mystery"
509,Safe,356.0,"Crime, Drama, Mystery"
514,The Last Dance,491.0,"Documentary, Biography, History"


In [54]:
df[df['RunTime']== 853.0]

,MOVIES,YEAR,GENRE,RATING,ONE-LINE,STARS,VOTES,RunTime,Gross,Year_start,end_year,is_ongoing,Stars,Director
1902,El tiempo entre costuras,(2013–2014),"Adventure, Drama, History",8.3,Sira Quiroga is a young Spanish dressmaker eng...,"\n \n Stars:\nAdriana Ugarte, \n...",3876.0,853.0,NaN,2013,2014,False,"Adriana Ugarte, \nMari Carmen Sánchez, \nTrist...",NaN


In [55]:
df['RunTime'] = df['RunTime'].fillna(df['RunTime'].median())

/var/folders/qw/_nk9z1hd74q9vjp80k85r5l80000gn/T/ipykernel_60558/199015607.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df['RunTime'] = df['RunTime'].fillna(df['RunTime'].median())


In [56]:
df['RunTime'].isnull().sum()

np.int64(0)

GROSS
Just drop this whole column — over 95% of it is missing.

In [57]:
df = df.drop(columns= 'Gross')

In [58]:
df

,MOVIES,YEAR,GENRE,RATING,ONE-LINE,STARS,VOTES,RunTime,Year_start,end_year,is_ongoing,Stars,Director
0,Blood Red Sky,(2021),"Action, Horror, Thriller",6.1,A woman with a mysterious illness is forced in...,\n Director:\nPeter Thorwarth\n| \n Star...,21062.0,121.0,2021,NaN,False,"Peri Baumeister, \nCarl Anton Koch, \nAlexande...",Peter Thorwarth
1,Masters of the Universe: Revelation,(2021– ),"Animation, Action, Adventure",5.0,The war for Eternia begins again in what may b...,"\n \n Stars:\nChris Wood, \nSara...",17870.0,25.0,2021,NaN,True,"Chris Wood, \nSarah Michelle Gellar, \nLena He...",NaN
2,The Walking Dead,(2010–2022),"Drama, Horror, Thriller",8.2,Sheriff Deputy Rick Grimes wakes up from a com...,"\n \n Stars:\nAndrew Lincoln, \n...",885805.0,44.0,2010,2022,False,"Andrew Lincoln, \nNorman Reedus, \nMelissa McB...",NaN
3,Rick and Morty,(2013– ),"Animation, Adventure, Comedy",9.2,An animated series that follows the exploits o...,"\n \n Stars:\nJustin Roiland, \n...",414849.0,23.0,2013,NaN,True,"Justin Roiland, \nChris Parnell, \nSpencer Gra...",NaN
4,Army of Thieves,(2021),"Action, Crime, Horror",7.1,"A prequel, set before the events of Army of th...",\n Director:\nMatthias Schweighöfer\n| \n ...,796.0,60.0,2021,NaN,False,"Matthias Schweighöfer, \nNathalie Emmanuel, \n...",Matthias Schweighöfer
...,...,...,...,...,...,...,...,...,...,...,...,...,...
9993,Totenfrau,(2022– ),"Drama, Thriller",7.1,Add a Plot,\n Director:\nNicolai Rohde\n| \n Stars:...,796.0,60.0,2022,NaN,True,"Felix Klare, \nRomina Küper, \nAnna Maria Mühe...",Nicolai Rohde
9995,Arcane,(2021– ),"Animation, Action, Adventure",7.1,Add a Plot,\n,796.0,60.0,2021,NaN,True,NaN,NaN
9996,Heart of Invictus,(2022– ),"Documentary, Sport",7.1,Add a Plot,\n Director:\nOrlando von Einsiedel\n| \n ...,796.0,60.0,2022,NaN,True,NaN,Orlando von Einsiedel
9997,The Imperfects,(2021– ),"Adventure, Drama, Fantasy",7.1,Add a Plot,\n Director:\nJovanka Vuckovic\n| \n Sta...,796.0,60.0,2021,NaN,True,"Morgan Taylor Campbell, \nIñaki Godoy, \nRhian...",Jovanka Vuckovic


In [59]:
df.columns

Index(['MOVIES', 'YEAR', 'GENRE', 'RATING', 'ONE-LINE', 'STARS', 'VOTES',
       'RunTime', 'Year_start', 'end_year', 'is_ongoing', 'Stars', 'Director'],
      dtype='object')

In [60]:
df.duplicated().sum()

np.int64(0)

In [61]:
df.isna().sum()

MOVIES           0
YEAR           501
GENRE            0
RATING           0
ONE-LINE         0
STARS            0
VOTES            0
RunTime          0
Year_start     579
end_year      8119
is_ongoing       0
Stars         1146
Director      4476
dtype: int64

In [62]:
df.columns

Index(['MOVIES', 'YEAR', 'GENRE', 'RATING', 'ONE-LINE', 'STARS', 'VOTES',
       'RunTime', 'Year_start', 'end_year', 'is_ongoing', 'Stars', 'Director'],
      dtype='object')

In [63]:
columns_to_keep = ['MOVIES','GENRE','Stars','Director','Year_start','end_year','is_ongoing','ONE-LINE','VOTES','RunTime', 'RATING',]

In [64]:
df = df[columns_to_keep]

In [65]:
df.columns

Index(['MOVIES', 'GENRE', 'Stars', 'Director', 'Year_start', 'end_year',
       'is_ongoing', 'ONE-LINE', 'VOTES', 'RunTime', 'RATING'],
      dtype='object')

In [66]:
df

,MOVIES,GENRE,Stars,Director,Year_start,end_year,is_ongoing,ONE-LINE,VOTES,RunTime,RATING
0,Blood Red Sky,"Action, Horror, Thriller","Peri Baumeister, \nCarl Anton Koch, \nAlexande...",Peter Thorwarth,2021,NaN,False,A woman with a mysterious illness is forced in...,21062.0,121.0,6.1
1,Masters of the Universe: Revelation,"Animation, Action, Adventure","Chris Wood, \nSarah Michelle Gellar, \nLena He...",NaN,2021,NaN,True,The war for Eternia begins again in what may b...,17870.0,25.0,5.0
2,The Walking Dead,"Drama, Horror, Thriller","Andrew Lincoln, \nNorman Reedus, \nMelissa McB...",NaN,2010,2022,False,Sheriff Deputy Rick Grimes wakes up from a com...,885805.0,44.0,8.2
3,Rick and Morty,"Animation, Adventure, Comedy","Justin Roiland, \nChris Parnell, \nSpencer Gra...",NaN,2013,NaN,True,An animated series that follows the exploits o...,414849.0,23.0,9.2
4,Army of Thieves,"Action, Crime, Horror","Matthias Schweighöfer, \nNathalie Emmanuel, \n...",Matthias Schweighöfer,2021,NaN,False,"A prequel, set before the events of Army of th...",796.0,60.0,7.1
...,...,...,...,...,...,...,...,...,...,...,...
9993,Totenfrau,"Drama, Thriller","Felix Klare, \nRomina Küper, \nAnna Maria Mühe...",Nicolai Rohde,2022,NaN,True,Add a Plot,796.0,60.0,7.1
9995,Arcane,"Animation, Action, Adventure",NaN,NaN,2021,NaN,True,Add a Plot,796.0,60.0,7.1
9996,Heart of Invictus,"Documentary, Sport",NaN,Orlando von Einsiedel,2022,NaN,True,Add a Plot,796.0,60.0,7.1
9997,The Imperfects,"Adventure, Drama, Fantasy","Morgan Taylor Campbell, \nIñaki Godoy, \nRhian...",Jovanka Vuckovic,2021,NaN,True,Add a Plot,796.0,60.0,7.1


In [67]:
df.isna().sum()

MOVIES           0
GENRE            0
Stars         1146
Director      4476
Year_start     579
end_year      8119
is_ongoing       0
ONE-LINE         0
VOTES            0
RunTime          0
RATING           0
dtype: int64

In [68]:
df[df.duplicated(keep=False)]

,MOVIES,GENRE,Stars,Director,Year_start,end_year,is_ongoing,ONE-LINE,VOTES,RunTime,RATING
6832,Mighty Little Bheem,"Animation, Short, Adventure",NaN,NaN,2019,NaN,True,Add a Plot,796.0,60.0,7.1
7656,Stranger Things,"Drama, Fantasy, Horror","David Harbour, \nFinn Wolfhard, \nMillie Bobby...",NaN,2016,NaN,True,Add a Plot,796.0,60.0,7.1
7764,Stranger Things,"Drama, Fantasy, Horror","David Harbour, \nFinn Wolfhard, \nMillie Bobby...",NaN,2016,NaN,True,Add a Plot,796.0,60.0,7.1
8418,Buddi,Animation,"Greta Jameson, \nRalf Jameson, \nClementine La...",NaN,2020,NaN,True,"The Buddis bounce, spin, glide - and giggle. -...",796.0,60.0,7.1
8419,Buddi,Animation,"Greta Jameson, \nRalf Jameson, \nClementine La...",NaN,2020,NaN,True,"The Buddis bounce, spin, glide - and giggle. -...",796.0,60.0,7.1
8424,Buddi,Animation,"Greta Jameson, \nRalf Jameson, \nClementine La...",NaN,2020,NaN,True,"The Buddis bounce, spin, glide - and giggle. -...",796.0,60.0,7.1
8716,Jaguar,"Drama, History","Blanca Suárez, \nIván Marcos, \nFrancesc Garri...",NaN,NaN,NaN,False,Add a Plot,796.0,60.0,7.1
8825,The Last Kingdom,"Action, Drama, History",NaN,NaN,2015,NaN,True,Add a Plot,796.0,60.0,7.1
8913,Jaguar,"Drama, History","Blanca Suárez, \nIván Marcos, \nFrancesc Garri...",NaN,NaN,NaN,False,Add a Plot,796.0,60.0,7.1
9032,Mighty Little Bheem,"Animation, Short, Adventure",NaN,NaN,2019,NaN,True,Add a Plot,796.0,60.0,7.1


In [69]:
df = df.drop_duplicates()

In [70]:
df.shape

(9470, 11)

In [71]:
data.shape

(9999, 9)

In [72]:
df.columns

Index(['MOVIES', 'GENRE', 'Stars', 'Director', 'Year_start', 'end_year',
       'is_ongoing', 'ONE-LINE', 'VOTES', 'RunTime', 'RATING'],
      dtype='object')

In [73]:
df.to_csv('Cleaned_movie_data.csv')

In [74]:
df.columns

Index(['MOVIES', 'GENRE', 'Stars', 'Director', 'Year_start', 'end_year',
       'is_ongoing', 'ONE-LINE', 'VOTES', 'RunTime', 'RATING'],
      dtype='object')